lib/Extension/LanguageServerPhpCsFixer/VersionResolver/PhpCsFixerVersionResolver.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerPhpCsFixer\VersionResolver;
+
+use Phpactor\Extension\LanguageServerPhpCsFixer\Model\PhpCsFixerProcess;
+use Phpactor\VersionResolver\SemVersion;
+use Phpactor\VersionResolver\SemVersionResolver;
+
+use Psr\Log\LoggerInterface;
+use function Amp\ByteStream\buffer;
+use function Amp\Promise\wait;
+
+class PhpCsFixerVersionResolver implements SemVersionResolver
+{
+    public function __construct(
+        private string $binPath,
+        private LoggerInterface $logger,
+    ) {
+    }
+
+    public function resolve(): ?SemVersion
+    {
+        $versionQuery = wait((new PhpCsFixerProcess($this->binPath, $this->logger))->run('--version'));
```

Comment:
In free time I need to investigate how such internal loops break `PhpCsFixerDiagnosticsProviderTest` and how we can fix it or at least work it around.